src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -15,13 +16,40 @@ namespace System.Device.Gpio.Drivers
     public class SysFsDriver : UnixDriver
     {
         private const string GpioBasePath = "/sys/class/gpio";
+        private const string GpioChip = "gpiochip";
+        private const string GpioLabel = "/label";
+        private const string GpioContoller = "pinctrl";
+        private const string GpioOffsetBase = "/base";
         private int _pollFileDescriptor = -1;
         private Thread _eventDetectionThread;
         private int _pinsToDetectEventsCount;
         private readonly static CancellationTokenSource s_eventThreadCancellationTokenSource = new CancellationTokenSource();
         private readonly List<int> _exportedPins = new List<int>();
         private readonly Dictionary<int, UnixDriverDevicePin> _devicePins = new Dictionary<int, UnixDriverDevicePin>();
         private readonly int _pollingTimeoutInMilliseconds = Convert.ToInt32(TimeSpan.FromMilliseconds(1).TotalMilliseconds);
+        private static int _pinOffset;
```

Comment:
this is a static field but you set it in the constructor - this doesn't sound right even if the read value will be the same. You should probably use static constructor or even better - field initializer

Reply:
just make it delay initialized.

Reply:
do you mean `int?` or `Lazy<int>`? I think field initializer will also lazy initialize on first usage

Reply:
both can work. 

> I think field initializer will also lazy initialize on first usage

That is true too. 

```C#
private static int _pinOffset = ReadOffset();

private static int ReadOffset()
{
    // write the reading code
}
```